sys/cddl/contrib/opensolaris/uts/common/dtrace/dtrace.c

```diff
@@ -6097,15 +6110,15 @@ dtrace_dif_emulate(dtrace_difo_t *difo, dtrace_mstate_t *mstate,
 	const char *strtab = difo->dtdo_strtab;
 	const uint64_t *inttab = difo->dtdo_inttab;
 
-	uint64_t rval = 0;
+	uintcap_t rval = 0;
 	dtrace_statvar_t *svar;
 	dtrace_dstate_t *dstate = &vstate->dtvs_dynvars;
 	dtrace_difv_t *v;
 	volatile uint16_t *flags = &cpu_core[curcpu].cpuc_dtrace_flags;
 	volatile uintptr_t *illval = &cpu_core[curcpu].cpuc_dtrace_illval;
 
 	dtrace_key_t tupregs[DIF_DTR_NREGS + 2]; /* +2 for thread and id */
-	uint64_t regs[DIF_DIR_NREGS];
+	uintcap_t regs[DIF_DIR_NREGS];
```

Comment:
There is another, extremely subtle thing that happens with registers and the `setx` instruction. I was unsure where to comment on it, so I will put it here.

```
dtrace -S -e -n 'syscall::read:entry { printf("%p\n", &`prison0); }'

DIFO 0x8009f0070 returns D type (pointer) (size 8)
OFF OPCODE      INSTRUCTION
00: 25000001    setx DT_INTEGER[0], %r1		! 0xffffffff81b0c7f0
01: 23000001    ret  %r1
```

Notice how ```&`prison0``` is a **pointer**, but is being created in the **integer table** as an integer. There is absolutely zero distinction between pointers and integers in the DTrace abstract machine, and this has to change in order for `setx` to work correctly.